tripper/dataset/dataset.py

```diff
@@ -0,0 +1,942 @@
+            as "_id=...".
+
+    Returns:
+        An updated copy of `dct`.
+
+    Notes:
+        The keys in `dct` and `kwargs` may be either properties defined in the
+        [JSON-LD context](https://raw.githubusercontent.com/EMMC-ASBL/oteapi-dlite/refs/heads/rdf-serialisation/oteapi_dlite/context/0.2/context.json)
+        or one of the following special keywords:
+          - "@id": Dataset IRI.  Must always be given.
+          - "@type": IRI of the ontology class for this type of data.
+            For datasets, it is typically used to refer to a specific subclass
+            of `emmo:DataSet` that provides a semantic description of this
+            dataset.
+
+    """
+    if "@id" not in dct:
+        raise ValueError("`dct` must have an '@id' key")
+
+    all_prefixes = get_prefixes()
+    if prefixes:
+        all_prefixes.update(prefixes)
+
+    d = prepare(type=type, dct=dct, prefixes=all_prefixes, kwargs=kwargs)
+
+    # Bind prefixes
+    for prefix, ns in all_prefixes.items():
+        ts.bind(prefix, ns)
+
+    # Write json-ld data to triplestore (using temporary rdflib triplestore)
+    f = io.StringIO(json.dumps(d))
+    with Triplestore(backend="rdflib") as ts2:
+        ts2.parse(f, format="json-ld")
+        ts.add_triples(ts2.triples())
+
+    # Add statements and data models to triplestore
+    save_extra_content(ts, d)
+
+    return d
+
+
+def save_extra_content(ts: Triplestore, dct: dict) -> None:
+    """Save extra content in `dct` to the triplestore.
+
+    Currently, this include:
+    - statements and mappings
+    - data models (require that DLite is installed)
+
+    """
+    # Save statements and mappings
+    statements = get_values(dct, "statements")
+    statements.extend(get_values(dct, "mappings"))
+    ts.add_triples(statements)
+
+    # Save data models
+    datamodels = get_values(dct, "datamodel")
+    try:
+        # pylint: disable=import-outside-toplevel
+        import dlite
+        from dlite.dataset import add_dataset
+    except ModuleNotFoundError:
+        if datamodels:
+            warnings.warn(
+                "dlite is not installed - data models will not be added to "
+                "the triplestore"
+            )
+    else:
+        for url in get_values(dct, "datamodelStorage"):
+            dlite.storage_path.append(url)
+
+        for uri in datamodels:
+            r = requests.get(uri, timeout=3)
+            if r.ok:
+                dm = dlite.Instance.from_json(r.content)
+                add_dataset(ts, dm)
+            else:
+                try:
+                    dm = dlite.get_instance(uri)
+                except (
+                    dlite.DLiteMissingInstanceError  # pylint: disable=no-member
+                ):
+                    # __FIXME__: check session whether want to warn or re-reise
+                    # in this case
+                    warnings.warn(f"cannot load datamodel: {uri}")
```

Comment:
```suggestion
```

Remove. This function should save the datadocumentation to the triplestore. It should not check whether the datamodel is available. And definitely not wheter it is a valid dlite datamodel.